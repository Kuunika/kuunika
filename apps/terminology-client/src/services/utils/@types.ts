export interface UIState {
  activePage: string;
}

export interface Category {
  categoryTitle: string;
  id: null | string | number;
  categories: Array<Category>;
}

export interface CategoryData {
  sourceHeadings: Array<string>;
  results: Array<any>;
  formatedData?: Array<any>;
}

export interface Concept {
  id: number | string | null;
  headings: Array<{ title: string; value: string }>;
  descriptions: Array<{ title: string; value: string }>;
  breadcrumb: string;
}
export interface IData {
  categories: Array<Category>;
  categoryData: { [key: string]: CategoryData };
  concept: { [key: string]: Concept };
}

export interface State {
  ui: UIState;
  data: IData;
  errors: any;
  loading: any;
}
