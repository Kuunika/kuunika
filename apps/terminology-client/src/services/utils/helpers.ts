import { Category } from './@types';

export const getSubCategories = (
  categories: Array<Category>,
  targetCategories = [] as Array<string>
) => {
  if (targetCategories.length === 0 || categories.length == 0)
    return categories;

  const targetCategory = categories.find(
    category =>
      category.categoryTitle.toLowerCase() ===
      targetCategories[0].toLowerCase().replace('-', ' ')
  );

  if (targetCategory.categories == null) return categories;

  return getSubCategories(
    targetCategory.categories,
    targetCategories.splice(1)
  );
};
