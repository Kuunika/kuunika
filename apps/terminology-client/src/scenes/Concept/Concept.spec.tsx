import React from 'react';
import { render } from '@testing-library/react';

import { ConceptView } from './index';
import { Concept as IConcept } from '../../services/utils/@types';

describe('Concept', () => {
  it('should render concept page successfully', () => {
    const data = {
      onBack: () => {},
      data: {
        id: 1,
        headings: [
          { title: 'title', value: 'value' },
          { title: 'new title', value: 'new value' }
        ],
        descriptions: [
          { title: 'des title', value: 'des value' },
          { title: 'new des title', value: 'new des value' }
        ],
        breadcrumb: 'clinical/ICD10'
      } as IConcept,
      breadCrumb: ['clinical', 'ICD10']
    };
    const { baseElement } = render(<ConceptView {...data} loading={false} />);
    expect(baseElement).toMatchSnapshot();
  });
});
