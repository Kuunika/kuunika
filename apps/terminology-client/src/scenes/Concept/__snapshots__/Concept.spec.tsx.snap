// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concept should render concept page successfully 1`] = `
<body>
  <div>
    <div
      class="sc-dnqmqq hWSScB"
    >
      <button
        class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-outlined"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <div
            class="sc-bxivhb fGSPuW"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-caret-left fa-w-6 "
              data-icon="caret-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 192 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
                fill="currentColor"
              />
            </svg>
          </div>
          BACK
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="sc-iwsKbI bSZgEP"
      >
        <h2
          class="sc-bdVaJa fWJwsq"
          data-testid="breadcrumb"
        >
          <b>
            Category: 
          </b>
          <span>
            <i
              class="sc-bwzfXH hCGXCx"
            >
              clinical
            </i>
            <i
              class="sc-htpNat diwqqW"
            />
          </span>
          <span>
            <i
              class="sc-bwzfXH hCGXCx"
            >
              ICD10
            </i>
          </span>
        </h2>
      </div>
      <div
        class="sc-ifAKCX dKvaeV"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
          >
            <span
              class="sc-EHOje klFhlA"
            >
              title
              : 
            </span>
            <span>
              value
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
          >
            <span
              class="sc-EHOje klFhlA"
            >
              new title
              : 
            </span>
            <span>
              new value
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-bZQynM gJHtkA"
      >
        <div
          class="sc-gzVnrw hWokKw"
        >
          Description
        </div>
        <div>
          <div
            class="sc-htoDjs kkNCdj"
          >
            des title
            : 
          </div>
          <p>
            des value
          </p>
        </div>
        <div>
          <div
            class="sc-htoDjs kkNCdj"
          >
            new des title
            : 
          </div>
          <p>
            new des value
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
